File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/transaction/TransactionPoolFactoryImpl.java
package org.palladiosimulator.blockchainsystems.core.transaction;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.TransactionPool;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.TransactionPoolFactory;

public class TransactionPoolFactoryImpl implements TransactionPoolFactory {

    @Override
    public TransactionPool createTransactionPool(String nodeId) {
        return new TransactionPoolImpl();
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/transaction/TransactionPoolFactoryImpl.kt
package org.palladiosimulator.blockchainsystems.core.transaction

import org.palladiosimulator.blockchainsystems.core.system.abstractions.TransactionPool
import org.palladiosimulator.blockchainsystems.core.system.abstractions.TransactionPoolFactory

/**
 * @author Davis Riedel
 */
class TransactionPoolFactoryImpl : TransactionPoolFactory {
  override fun createTransactionPool(nodeId: String): TransactionPool {
    return TransactionPoolImpl()
  }
}